repository_service_tuf_worker/repository.py

```diff
@@ -244,7 +245,12 @@ def _persist(self, role: Metadata, role_name: str) -> str:
             if filename[0].isdigit() is False:
                 filename = f"{role.signed.version}.{filename}"
 
-        role.to_file(filename, JSONSerializer(), self._storage_backend)
+        bytes_data = role.to_bytes(JSONSerializer())
+
+        with tempfile.TemporaryFile() as temp_file:
+            temp_file.write(bytes_data)
+
+            self._storage_backend.put(temp_file, filename)
```

Comment:
Can you explain why we need a temporary file to send to backend storage?

Reply:
Honestly, I just followed what python-tuf did with their `Metadata.to_file()` function: 
https://github.com/theupdateframework/python-tuf/blob/cc48e9d5f158022405349fe9854d0ae67c11ae53/tuf/api/metadata.py#L337-L339
The reason I can think of why `python-tuf` used that from the start is that they decided to use `tempFile` to make sure that if there is an error with a file name already existing they won't overwrite the old file.

Reply:
We can add such a check and remove the `tempFile` usage.

Reply:
Together with @kairoaraujo we discussed that there is no sense in writing the file inside the `_persist` method and instead we can move it into the `LocalStorage` implementation. 
Also, we decided to remove the usage of the `temp` module.